Library/EnvelopeFactory.ts

```diff
@@ -1,13 +1,163 @@
+import os = require("os");
 import Contracts = require("../Declarations/Contracts")
 import Util = require("./Util")
 import Config = require("./Config");
 import Context = require("./Context");
 import { CorrelationContextManager } from "../AutoCollection/CorrelationContextManager";
 
+var StreamId = Util.w3cTraceId(); // Create a guid
+
 /**
  * Manages the logic of creating envelopes from Telemetry objects
  */
 class EnvelopeFactory {
+    public static createQuickPulseEnvelope(metrics: Contracts.MetricQuickPulse[], documents: Contracts.DocumentQuickPulse[], config: Config, context: Context): Contracts.EnvelopeQuickPulse {
+        const machineName = os.hostname(); // Note: os.hostname() was added in node v0.3.3
+        const instance = (context.tags
+            && context.keys
+            && context.keys.cloudRoleInstance
+            && context.tags[context.keys.cloudRoleInstance]) || machineName;
+
+        var envelope: Contracts.EnvelopeQuickPulse = {
+            Documents: documents.length > 0 ? documents : null,
+            InstrumentationKey: config.instrumentationKey || "",
```

Comment:
Is it possible to not have an ikey?

Reply:
I don't think so. I carried it over from the other envelopecreator stuff, but it probably doesn't need to be there either